Add TLS, setup details and Talk 12 changes to TURN server documentation (#5788)
nextcloud/spreed: docs/TURN.md

- Nextcloud Talk will try direct P2P in the first place, use STUN if needed and TURN as last resort fallback. Thus to be most flexible and guarantee functionality of your Nextcloud Talk instance in all possible connection cases, you would want to setup a TURN server.

#### TURN server and Nextcloud Talk High Performance Backend

A TURN server might be needed even if the Nextcloud Talk High Performance Backend is used and it is publicly accessible.

> The High Performance Backend uses a certain range of ports for WebRTC media connections (20000-40000 by default). A client could be behind a restrictive firewall that only allows connections to port 443, so even if the High Performance Backend is publicly accessible the client would need to connect to a TURN server in port 443, and the TURN server will then relay the packets to the 20000-4000 range in the High Performance Backend.

Reviewer: ```suggestion
The High Performance Backend uses a certain range of ports for WebRTC media connections (20000-40000 by default). A client could be behind a restrictive firewall that only allows connections to port 443, so even if the High Performance Backend is publicly accessible the client would need to connect to a TURN server in port 443, and the TURN server will then relay the packets to the 20000-40000 range in the High Performance Backend.
```

no-multicast-peers
```

- Support for TLS connections to the TURN server has been added in Talk 11. In some cases clients can be behind very restrictive firewalls that only allow TLS connections; in those cases the clients would be able to connect with other clients or the High Performance Backend only through a TURN server and a TLS connection. However, please note that TLS connections do not provide any additional security, as media streams are always end-to-end* encrypted in WebRTC; enabling TLS is just a matter of providing the maximum compatibility.

>   *When the High Performance Backend is used the High Performance Backend is one of the ends; in that case the media streams are not currently encrypted end-to-end at the client level.

Reviewer: ```suggestion
  *When the High Performance Backend is used the High Performance Backend is one of the ends; in that case the media streams are not end-to-end encrypted between the users but only between users and the High Performance Backend.
```

Author: Done, although I replaced _users_ with _participants_.
